src/client/visualizations/data-provider/data-provider.tsx

```diff
@@ -39,9 +41,10 @@ export type QueryFactory = Binary<Essence, Timekeeper, Expression>;
 interface DataProviderProps {
   refreshRequestTimestamp: number;
   essence: Essence;
-  timekeeper: Timekeeper;
+  appSettings: ClientAppSettings;
+  timekeeper: Timekeeper; // TODO: check if needed
   stage: Stage;
-  queryFactory: QueryFactory;
+  queryFactory: QueryFactory; // TODO: remove later
```

Comment:
That's not needed, we pick query factory function on backend, based on visualisation name.